src/WSRequestHandler_ComponantsManagement.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+* obs-websocket
+* Copyright (C) 2019	Ilyes KAANICH <https://github.com/ilyes64>
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <https://www.gnu.org/licenses/>
+*/
+#include <QtCore/QString>
+#include <QtCore/QBuffer>
+#include <QtCore/QFileInfo>
+#include <QtGui/QImage>
+#include <QtGui/QImageWriter>
+
+#include "Utils.h"
+
+#include "WSRequestHandler.h"
+/**
+* Create a new source
+*
+* @param {String} `sourceName` Name of the source to be created
+* @param {String} `typeIDName` type of the source to be created
+* @param {Object} `sourceSettings` New settings. These will be merged to the new created source.
+* @param {Object} `hotkeyData` hotkeyData. These will be merged to the new created source.
+*
+* @api requests
+* @name AddNewSource
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleCreateNewSource(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceName") || !req->hasField("typeIDName") || !req->hasField("sourceSettings")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sourceName = obs_data_get_string(req->data, "sourceName");
+	const char* typeIDName = obs_data_get_string(req->data, "typeIDName");
+	
+	OBSDataAutoRelease newsourceSettings = obs_data_get_obj(req->data, "sourceSettings");
+	OBSDataAutoRelease hotkeyData = obs_data_get_obj(req->data, "hotkeyData");
+	
+	obs_source_t* source = obs_source_create(typeIDName, sourceName, newsourceSettings, hotkeyData);
```

Comment:
Please check if the source was created successfully (source != nullptr). If not the case, return an error.